laravel/tests/Unit/AutoDeliveryServiceTest.php

```diff
@@ -25,4 +31,106 @@ public function testNoOrderSendMail()
         AutoDeliveryService::NoOrderSendMail($ship_date);
         Mail::assertSent(AutoDeliverySystemNotification::class, 1);
     }
+
+    use RefreshDatabase;
+
+    /**
+     * @test
+     * @expectedExceptionMessage
+     */
+    public function testTryOrderItemsAndInventoriesNoneAll()
+    {
+        //在庫が一つも無いパターン
+        $ship_date = '2020-01-01'; 
+        $order_item = factory(OrderItem::class, 'test_order_mock_date_1')->make([
+                                'item_code' => 'test111',
+                                'ship_date' => $ship_date
+                                ]);
+
+        $this->expectExceptionMessage("在庫全く無し");
+
+        AutoDeliveryService::TryOrderItemsAndInventories($order_item);
+    }
+
+    /**
+     * @test
+     * @expectedExceptionMessage
+     */
+    public function testTryOrderItemsAndInventoriesALittle()
+    {
+        //注文が足りないパターン
+        $this->seed();
+
+        $ship_date = '2020-01-02'; 
+
+        $order_item = factory(OrderItem::class, 'test_order_mock_date_2')->make([
+                                'done_flag' => 0,
+                                'weight' => 2000000,
+                                'ship_date' => $ship_date
+                                ]);
+
+        //　TODO 出来ればモックを使いたい
+
+        $inventories = Inventory::all();
+
+        $factory_stock = \Config::get('const.Constant.factory_stock');
+        $warehouse_stock = \Config::get('const.Constant.warehouse_stock');
+
+        /*　この方法だとエラー
+        $inventory = $inventories->where('item_code', $order_item->item_code)
+                    ->sortBy('charge_code')
+                    ->sortBy('order_code')
+                    ->sortBy('manufacturing_code')
+                    ->sortBy('bundle_number')
+                    ->last();*/
+
+        $inventory = $inventories->where('item_code', $order_item->item_code)
+                    ->sortByDesc('bundle_number')
+                    ->sortByDesc('manufacturing_code')
+                    ->sortByDesc('order_code')
+                    ->sortByDesc('charge_code')
+                    ->first();
```

Comment:
実際の処理ではModelのScopeを呼び出しているので、ここでもscopeを呼び出してデータを取得すれば良いかなと思いました。
```suggestion
$inventory = Inventory::SearchByItemCodeAndStatus($order_item)->get()->last();
```